### PR TITLE
Protein translation

### PR DESCRIPTION
**Pull request by RobinCsl**
Same comment as for PR #364, I did some kind of manual rebase so that there shouldn't be any conflicts. Looking forward to having your comments.

### CODE REVIEW

File: exercises/protein-translation/protein-translation.spec.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  xtest('Small RNA strand', () => {
+    expect(translate('AUGUUUUCU')).toEqual(['Methionine', 'Phenylalanine', 'Serine']);
+  });
+
+  xtest('Stop codon ends translation', () => {
+    expect(translate('AUGUUUUCUUAAAUG')).toEqual(['Methionine', 'Phenylalanine', 'Serine']);
+  });
+
+  xtest('Invalid codon throws error', () => {
+    expect(() => { translate('LOL'); }).toThrow(new Error('Invalid codon'));
```
Comment (matthewmorgan) [inline exercises/protein-translation/protein-translation.spec.js]: We could clean this up a little like so:
```javascript
expect(() => translate('LOL')).toThrow(new Error('Invalid codon'));
```

File: exercises/protein-translation/example.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    case 'UAA':
+    case 'UAG':
+    case 'UGA':
+      return 'STOP';
+
+    default:
+      return 'INVALID';
+  }
+};
```
Comment (matthewmorgan) [inline exercises/protein-translation/example.js]: Nothing at all wrong with the massive `switch` statement, but I sometimes like an alternative approach to these sorts of things, using an object.  EG:

```javascript
const ACID_PROTEIN_MAP = {
  'AUG': 'Methionine',
  'UUU': 'Phenylalanine',
  'UUC': 'Phenylalanine',
// and so on...
};

const getProtein = codon => ACID_PROTEIN_MAP[codon] || 'INVALID';
```

It remains very readable and gets what's essentially configuration out of your logic.  In production, you might pull the config out even further: this makes it easy to adjust the mapping without having to redeploy code.

**Comment by RobinCsl** [comment]
Implemented the changes proposed. I tried to extract the config in an external json file, but it seemed to cause more complications, so I finally opted for the easier solution of setting this variable in the same file.

File: exercises/protein-translation/acid-protein-map.config.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "UCG": "Serine",
+  "UAU": "Tyrosine",
+  "UAC": "Tyrosine",
+  "UGU": "Cysteine",
+  "UGC": "Cysteine",
+  "UGG": "Tryptophan",
+  "UAA": "STOP",
+  "UAG": "STOP",
+  "UGA": "STOP"
+}
```
Comment (matthewmorgan) [inline exercises/protein-translation/acid-protein-map.config.json]: Gorgeous!  Awesome indeed.

**Comment by matthewmorgan** [comment]
@RobinCsl I see why you're having a problem-- I think the makefile that runs the test suite does not copy your config file over...not a problem to include it in the example file.

It's nice, though, right?  The config is ~ 1/2 the size of the `switch` statement and just as clear.

This PR looks good to me, if you could please squash the commits we'll merge it.

**Comment by rchavarria** [comment]
Extract configuration to external files is usually a good idea in the real world. In exercism we try to get just three files: readme, tests and production/solution. Keeping all tests and all solution code in one file each make solution management much easy.

I think that could be the reason why the `Makefile` doesn't take care of other files. It *only* knows about the files described above. 

Sorry @RobinCsl for causing you some struggles with that. And **thanks** for implementing this exercise for the track.

**Comment by RobinCsl** [comment]
I tried to squash all the commits together, but every time I push there would still be a conflict with master, even though I rebased off upstream/master. So I believe I will need assistance with squashing my commits. Sorry!

**Comment by RobinCsl** [comment]
Looks like I managed now, I guess I had not rebased in the right way.

**Comment by rchavarria** [comment]
Sorry :disappointed: I think merging PRs #362 and #364 created conflicts in this one.

**Comment by RobinCsl** [comment]
No problem, should be fine now :) Thank you!

**Comment by matthewmorgan** [comment]
@RobinCsl glad to hear you got it sorted.  I did not see you were having trouble, hope it wasn't too frustrating.  Sometimes it can be a bit tricky to resolve the conflicts.  

Thanks for the contribution!

**Pull request merged**